lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -1257,6 +1257,10 @@ RequirementCheckResult TypeChecker::checkGenericArguments(
         // pass it on up.
         return status;
       case RequirementCheckResult::Success:
+        // FIXME: we should possibly have a queue of requirements (or
+        // ArrayRef<Requirements>) that this function works through, and add the
+        // conditional requirements to the end of this.
```

Comment:
Yeah, that makes sense. You're discovering additional requirements along the way that'll need to be checked. The "listener" interface might be a little tricky.